etcdserver/snapshot_store.go

```diff
@@ -67,7 +83,7 @@ func newSnapshotStore(dir string, kv dstorage.KV) *snapshotStore {
 // getSnap returns a snapshot.
 // If there is no available snapshot, ErrSnapshotTemporarilyUnavaliable will be returned.
 func (ss *snapshotStore) getSnap() (*snapshot, error) {
-	if ss.snap != nil {
+	if ss.snap != nil && !ss.snap.isClosed() {
```

Comment:
the code is correct, but it is not straightforward when reading the code. If snapshotStore has some open snapshot still, it cannot request new snapshot and return no available snapshot. I suppose that adding some comments could help to understand.